### PR TITLE
Valid hostname check

### PR DESCRIPTION
**Pull request by sthulb**
Allowed hostname chars are: 0-9a-zA-Z . -

Refs: #383

### CODE REVIEW

File: host.go [hunk 1]
```diff
@@ -78,7 +80,7 @@ func LoadHost(name string, storePath string) (*Host, error) {
 
 func ValidateHostName(name string) (string, error) {
```
Comment (ehazlett) [inline host.go]: This seems like we could get some easy tests added for this?  Would you mind? :)

Comment (ehazlett) [inline host.go]: Otherwise looks good!  Thanks!

Comment (sthulb) [inline host.go]: Sure, not a problem.

**Comment by sthulb** [comment]
I've added some tests :)

**Comment by ehazlett** [comment]
LGTM

**Pull request merged**